RecoMuon/MuonRechitClusterProducer/plugins/CSCRechitClusterProducer.cc

```diff
@@ -0,0 +1,237 @@
+      float x = globalPosition.x();
+      float y = globalPosition.y();
+      float z = globalPosition.z();
+      rechitRef csc_ref = rechitRef(cscRechits, recIt - cscRechits->begin());
+      inputs_.push_back(csc_ref);
+      fjInput.push_back(fastjet::PseudoJet(x, y, z, 1.0));
+      fjInput.back().set_user_index(recIt - cscRechits->begin());
+    }
+  }
+  fastjet::ClusterSequence clus_seq(fjInput, jet_def);
+
+  double ptmin = 0.0;
+  std::vector<fastjet::PseudoJet> fjJets = fastjet::sorted_by_pt(clus_seq.inclusive_jets(ptmin));
+
+  auto CSCclusters = std::make_unique<MuonCSCRecHitClusterCollection>();
+  if (!fjJets.empty()) {
+    for (unsigned int ijet = 0; ijet < fjJets.size(); ++ijet) {
+      // get the fastjet jet
+      const fastjet::PseudoJet& fjJet = fjJets[ijet];
+
+      // skip if the cluster has too few rechits
+      if (int(fjJet.constituents().size()) < nRechitMin_)
+        continue;
+      // get the constituents from fastjet
+      std::vector<fastjet::PseudoJet> const& fjConstituents = fastjet::sorted_by_pt(fjJet.constituents());
+
+      std::vector<rechitRef> const& rechits = getConstituents(fjConstituents);
+
+      //Derive cluster properties
+      float time = 0.0;
+      float time_strip = 0.0;  // for timeSpread calculation
+      int nME11 = 0;
+      int nME12 = 0;
+      int nStation10 = 0;
+      int totStation10 = 0;
+      float avgStation10 = 0.0;
+      float timeSpread = 0.0;
+      std::map<int, int> station_count_map;
+      for (auto& rechit : rechits) {
+        CSCDetId cscdetid = rechit->cscDetId();
+        int endcap = CSCDetId::endcap(cscdetid) == 1 ? 1 : -1;
+        int stationRing = endcap * (CSCDetId::station(cscdetid) * 10 + CSCDetId::ring(cscdetid));
+        if (CSCDetId::ring(cscdetid) == 4)
+          stationRing = endcap * (CSCDetId::station(cscdetid) * 10 + 1);  // ME1/a has ring==4
+        station_count_map[CSCDetId::station(cscdetid)]++;
+        if (abs(stationRing) == 11)
+          nME11++;
+        if (abs(stationRing) == 12)
+          nME12++;
+        time += (rechit->tpeak() + rechit->wireTime());
+        time_strip += rechit->tpeak();
+      }
+      //station statistics
+      std::map<int, int>::iterator it;
+      for (it = station_count_map.begin(); it != station_count_map.end(); it++) {
+        if (it->second >= 10) {
+          nStation10++;
+          avgStation10 += (it->first) * (it->second);
+          totStation10 += (it->second);
+        }
+      }
+      if (nStation10 != 0) {
+        avgStation10 = avgStation10 / totStation10;
+      }
+      time = time / (2 * rechits.size());
+      time_strip = time_strip / (rechits.size());
+
+      //derive cluster statistics
+      for (auto& rechit : rechits) {
+        timeSpread += (rechit->tpeak() - time_strip) * (rechit->tpeak() - time_strip);
+      }
+      timeSpread = std::sqrt(timeSpread / rechits.size());
+
+      float jetX = fjJet.px() / rechits.size();
```

Comment:
```suggestion
      float invN = 1.f/rechits.size();
      float jetX = fjJet.px() * invN;
```
also, use this invN in the cases above